R/rdeephaven/R/agg_ops_wrapper.R

```diff
@@ -1,6 +1,73 @@
+#' allow you to aggregate tables on a per-group basis.
+#'
+#' @section
+#' Apply aggregations to a table:
+#' There are two methods for performing aggregations on a table, `agg_by()` and `agg_all_by()`. `agg_by()` allows you to
+#' perform many aggregations on specified columns, while `agg_all_by()` allows you to perform a single aggregation to
+#' every column in the table. Both methods have an optional `by` parameter that is used to specify grouping columns.
+#' Here are some details on each method:
+#'
+#' - `TableHandle$agg_by(aggs, by)`: Creates a new table containing grouping columns and grouped data.
+#'   The resulting grouped data is defined by the aggregation(s) specified.
+#' - `TableHandle$agg_all_by(agg, by)`: Creates a new table containing grouping columns and grouped data.
+#'   The resulting grouped data is defined by the aggregation specified. This method applies the aggregation to all
+#'   columns of the table, so it can only accept one aggregation at a time.
+#'
+#' The `agg_by()` and `agg_all_by()` methods themselves do not know anything about the columns you want to perform
+#' aggregations on. Rather, the desired columns are passed to individual `agg` functions, enabling you to apply
+#' various kinds of aggregations to different columns or groups of columns as needed.
+#'
+#' @section
+#' `agg` functions:
+#' `agg` functions are used to perform aggregation calculations on grouped data by passing them to `agg_by()` or
+#' `agg_all_by()`. These functions are _generators_, meaning they return _functions_ that the Deephaven engine knows
+#' how to interpret. We call the functions that they return `AggOp`s. These `AggOp`s are not R-level functions, but
+#' wrappers around C++ functions that perform all of the intensive calculations. Here is a list of all `agg` functions
+#' available in Deephaven:
+#'
+#' - [`agg_first()`][agg_first]
+#' - [`agg_last()`][agg_last]
+#' - [`agg_min()`][agg_min]
+#' - [`agg_max()`][agg_max]
+#' - [`agg_sum()`][agg_sum]
+#' - [`agg_abs_sum()`][agg_abs_sum]
+#' - [`agg_avg()`][agg_avg]
+#' - [`agg_w_avg()`][agg_w_avg]
+#' - [`agg_median()`][agg_median]
+#' - [`agg_var()`][agg_var]
+#' - [`agg_std()`][agg_std]
+#' - [`agg_percentile()`][agg_percentile]
+#' - [`agg_count()`][agg_count]
+#'
+#' For more details on each aggregation function, click on one of the methods above or see the reference documentation
+#' by running `?agg_first`, `?agg_last`, etc.
+#'
+#' @section
+#' Examples:
+#'
+#' ```
+#' NULL
+#' ```
+#'
+NULL
+
+# An AggOp represents an aggregation operator that can be passed to agg_by() or agg_all_by(). This is the return type
+# of all of the `agg` functions. It is a wrapper around an Rcpp_INTERNAL_AggOp, which itself is a wrapper around a
+# C++ AggregateWrapper, which is finally a wrapper around a C++ Aggregate. See rdeephaven/src/client.cpp for details.
+# Note that AggOps should not be instantiated directly by user code, but rather by provided `agg` functions.
```

Comment:
These are regular comments and not Roxygen comments. They don't get turned into documentation. This is here for the next dev reading the source code, because it's kind of convoluted if you haven't seen anything like it before.